src/filesystem.cpp

```diff
@@ -759,23 +783,19 @@ static bfs::path apple_userdata(const std::string& newprefdir)
 		DBG_FS << "userdata using SDL pref path";
 #else
 		temp = "Library/Application Support/Wesnoth_"+get_version_path_suffix();
-		DBG_FS << "userdata using default path relative to HOME";
+		DBG_FS << "Using default relative path for userdata";
 #endif
-	} else if(temp[0] != '/') {
-		// TRANSLATORS: translate the part inside <...> only
-		deprecated_message(_("--userdata-dir=<relative path>"),
-			DEP_LEVEL::FOR_REMOVAL,
-			{1, 17, 0},
-			_("Use absolute paths. Relative paths are deprecated because they are interpreted relative to $HOME"));
 	}
 
 	// if it's an absolute path, just use that
 	// else make it relative to HOME if HOME is populated, otherwise make it relative to the current working directory
 	if(temp[0] == '/') {
 		dir = temp;
+		DBG_FS << "userdata using absolute path: " << temp;
 	} else {
-		bfs::path home = home_str ? home_str : ".";
+		bfs::path home = home_str ? home_str : get_cwd();
```

Comment:
Macos still falls back to HOME.

Reply:
It has to, since that's what it expects to do elsewhere (ie `migrate_apple_config_directory_for_unsandboxed_builds()`).

Reply:
The migrate function expects the default dir. That should not change.

Reply:
Changed back in 2991c389550dd13dca8263adf9414b7876d4aabe then

Reply:
I guess I didn't express what I wanted to say. As far as I can tell nothing changed so far. Whether the current dir is written as `.` or the absolute path of the current dir makes no difference besides cosmetics.

Perhaps it's not clear what that migrate function does. It doesn't really migrate anything. It makes a symlink from the location of a "normal" wesnoth to the location of a sandboxed wesnoth if that dir exists.
1) If HOME is not set this will not work at all. (Presumably there is a proper way to get the sandbox dir without using HOME but I don't think we need to care much.)
2) If the "normal" wesnoth is run first, it will not work either since it will create dirs as usual and if a sandboxed wesnoth is run later and then again a normal wesnoth it's too late to make a symlink.

It's also not really that important since I imagine it's rare for someone to run sandboxed and unsandboxed wesnoth of the same version interchangeably. Probably just going to happen for a dev.

Reply:
Well, the strange thing is that it seems (?) that despite the symlinks it's still actually setting the userdata dir to the non-sandboxed folder. Which never made sense to me since the sandboxing came later, so shouldn't it be writing to the sandboxed location instead of the non-sandboxed location? Also it's kind of weird that the sandboxing even allows that to work in the first place given the whole point of sandboxing is to prevent an application from accessing external locations/resources.

Reply:
Presumably the sandboxing is implemented by diverting the standard location to the sandboxed one. Otherwise you'd have to rewrite every program.